### PR TITLE
Restart GDB

### PR DESCRIPTION
**Pull request by rstyd**
This PR addresses #704.

### CODE REVIEW

File: beeflow/common/db/wfm_db.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def __init__(self, db_file):
         """Initialize Task, db_file, and Workflow object."""
         self.Task = namedtuple("Task", "id task_id workflow_id name resource state slurm_id") #noqa
         self.db_file = db_file
-        self.Workflow = namedtuple("Workflow", "id workflow_id name state run_dir bolt_port gdb_pid") #noqa
+        self.Workflow = namedtuple("Workflow", "id workflow_id name state run_dir bolt_port http_port, https_port, gdb_pid") #noqa
```
Comment (jtronge) [inline beeflow/common/db/wfm_db.py]: Should commas be between all of those? I guess it's just stripping out the extra commas for http_port and https_port.

File: beeflow/wf_manager/resources/wf_utils.py [hunk 2]
```diff
@@ -123,16 +125,42 @@ def create_wf_namefile(wf_name, wf_id):
...
+            # Returns an exception if the GDB is running
+            os.kill(gdb_pid, 0)
+            # The GDB process is running so we'll kill it and restart it
+        except ProcessLookupError:
+            # The GDB process isn't running. Restart it
+            log.error('The GDB is currently down restarting it')
+            wf_dir = get_workflow_dir(wf_id)
+            http_port = db.workflows.get_http_port(wf_id)
+            https_port = db.workflows.get_https_port(wf_id)
+            gdb_pid = dep_manager.start_gdb(wf_dir, bolt_port, http_port, https_port)
```
Comment (jtronge) [inline beeflow/wf_manager/resources/wf_utils.py]: Will this get activated by a user request (i.e. a beeflow query)? I wonder if we should launch this as a celery task.

**Comment by jtronge** [comment]
I tried this with `examples/clamr-ffmpeg-build`. I killed neo4j while the clamr task was running and it looks like the WFM restarted neo4j. I think the clamr task completed but the state is still at RUNNING and it doesn't get to the ffmpeg step.

**Comment by rstyd** [comment]
Oh I'll check that out. Thanks! I bet the code that normally sets that is getting short skipped.

**Comment by pagrubel** [comment]
> I tried this with `examples/clamr-ffmpeg-build`. I killed neo4j while the clamr task was running and it looks like the WFM restarted neo4j. I think the clamr task completed but the state is still at RUNNING and it doesn't get to the ffmpeg step.



> Oh I'll check that out. Thanks! I bet the code that normally sets that is getting short skipped.

You may also need to look at what commands you are using to check the status. I noticed in the past if a job finishes the status might not be available on some systems.

**Comment by pagrubel** [comment]
@rstyd There have been several merges into develop so please merge develop in this branch

**Comment by jtronge** [comment]
This works for me. I tried the clamr-ffmpeg-build workflow and killed the gdb twice, during the clamr and then the ffmpeg step, and both times it was restarted.

@rstyd Can you merge/rebase on top of develop to check against the recent neo4j updates? I'll approve after that.

**Comment by pagrubel** [comment]
@rstyd I also tried this and it works for me. Would you please make a squash commit with the important description of what was done. There have been many changes to develop since you started.

**Comment by rstyd** [comment]
No problem!

**Comment by pagrubel** [comment]
Also, reexcute works well, thanks for fixing that!!!

**Review by jtronge — APPROVED** [review]
Looks great! I tested the clamr workflow and bee had no problems restarting neo4j.

**Pull request merged**